### DISCUSSION TITLE
How to link an external library?

### DISCUSSION THREAD

**Discussion by ghost** [modula3/cm3#1078] [General]
It seems this must be via m3makefile too but I can't find any examples.

**Comment by mikanystrom** [comment]
https://modula3.elegosoft.com/cm3/doc/tutorial/m3/m3_6.html

import_lib

I generally don't do it this way, because it makes
the m3makefile non-portable.  Instead I add the libraries directly to the
config, so that the libraries can reside in different places on different
machines.  You could do something tricky with includes or write Quake I
guess if you wanted to automate that.

But if you are working on a single system, yes you can use import_lib.

     Mika
